### ISSUE TITLE
"Forgot password" does not work

### ISSUE THREAD

**Issue by mitchwagner** [Murali-group/GraphSpace#298]
This should be a top priority. Without this, users unable to log in are at risk of never returning to the site.

**Comment by tmmurali** [comment]
@mitchwagner Please provide more details, including what happens (or does not) when you use the "Forgot password" link.

**Comment by mitchwagner** [comment]
The page currently appears as follows:

![page](https://user-images.githubusercontent.com/5316207/28875054-2f138f34-7762-11e7-9e67-bfe9894ce37b.png)

Upon entering one's email and clicking the "send email" button, the user is taken to this page:

![error2](https://user-images.githubusercontent.com/5316207/28875056-3270d9ca-7762-11e7-87ff-b9edc5555ea7.png)

To my understanding, this has been the status of this button for several months.

Furthermore, the red "close" button on the first of the above screenshots does not seem to evoke any action, at least on Mozilla Firefox 54.0 (64-bit) on Ubuntu 16.

**Comment by jlaw9** [comment]
I was able to reset my password just find using my email username. I was also able to re-create this issue by putting a non-email in the email field (such as 't').

This issue seems to be a result the fact that we do not require an email as a username (issue #297)

**Comment by mitchwagner** [comment]
I stand corrected. I was operating under the assumption that I had created an account on GraphSpace.org with the email I used, but it must have been the test server.

What this indicates then is that GraphSpace displays what appears to be an error to the user if they enter in an email that is not stored with the system. So, in addition to #297, it might be better to display a "user not found" message, or something along the lines of "If so and so user has an account with GraphSpace.org, an email will arrive in their inbox shortly," if we don't want to leak information about who has an account.
